### PR TITLE
Use Workspace Dependencies #2406

### LINKED ISSUE #2406: Use workspace dependency
https://doc.rust-lang.org/cargo/reference/workspaces.html#the-dependencies-table

### PR DESCRIPTION
**Pull request by abigailajohn**
This PR centralizes shared dependencies to 
the workspace level, ensuring version 
consistency and simplifying dependency 
management across all crates.

Changes:
-  Introduced workspace.dependencies in the 
main Cargo.toml for shared dependencies used 
in two or more crates.
- Adjusted crate-specific Cargo.toml files to 
reference shared dependencies from the workspace.

Testing:
Ran cargo build successfully with no errors
All tests passed with cargo test

resolves [#2406](https://github.com/nmstate/nmstate/issues/2406)

### CODE REVIEW

**Review by cathay4t — CHANGES_REQUESTED** [review]
Could you move all crate dependency to top workspace level so we have single place for dependencies.

**Comment by abigailajohn** [comment]
> Could you move all crate dependency to top workspace level so we have single place for dependencies.

could you clarify if by "all crate dependency" you are referring to every single dependency including those used by only one crate, or just the shared dependencies?

**Comment by cathay4t** [comment]
every crates

File: rust/Cargo.toml [hunk 1]
```diff
@@ -6,6 +6,15 @@ members = [
     "src/lib",
 ]
 
+[workspace.dependencies]
+log = "0.4.17"
+serde_yaml = "0.9"
+serde = {version = "1.0.137", default-features = false, features = ["derive"]}
+serde_json = { version = "1.0.75", default-features = false }
+nmstate = { path = "src/lib", default-features = false }
```
Comment (cathay4t) [inline rust/Cargo.toml]: Version is missing, it is needed for uploading to crates.io

**Review by cathay4t — CHANGES_REQUESTED** [review]
Nix zbus are not found in workspace yet.

**Comment by abigailajohn** [comment]
> Nix zbus are not found in workspace yet.

I have added them to the workspace and also added the nmstate version.

**Review by cathay4t — CHANGES_REQUESTED** [review]
There are still leftover!

**Review by cathay4t — APPROVED** [review]

**Pull request merged**